4844: Specifications for bytecode, codehash & gas limits of Ectone Upgrade Txns

danyalprout opened the pull request:
**Description**

For the Ecotone hardfork, specify the transaction data for the GasPriceOracle and L1Block transactions. You can sanity check this by either running the commands or looking at the bindings.

* [L1Block](https://github.com/ethereum-optimism/optimism/blob/5996d0bc1a4721f2169ba4366a014532f31ea932/op-bindings/bindings/l1block.go#L34)
* [GasPriceOracle](https://github.com/ethereum-optimism/optimism/blob/5996d0bc1a4721f2169ba4366a014532f31ea932/op-bindings/bindings/gaspriceoracle.go#L34C8-L34C2758)

danyalprout committed:
Update specs to reflect literal bytecode

danyalprout committed:
Move bytecode to separate files

tynes commented:
It wouldn't be a bad idea to include the codehash (of the deployed bytecode) in the spec so that we could write an assertion against it or we could just do a bytes comparison against the code directly

danyalprout committed:
Add code hash to the specs

danyalprout committed:
Specify the gas limits for upgrade txns

danyalprout commented:
Also added some specification around the upgrade transaction gas limits for https://github.com/ethereum-optimism/protocol-quest/issues/62.

I ran these txns as part of the e2e tests and captured the amount they used and added a buffer:

| Txn | e2e Tests | Proposal |
|--------|--------|--------|
| Deploy L1Block | 341,000 | 375,000 |
| Deploy GasPriceOracle | 916,000 | 1,000,000 |
| Update L1Block Proxy | 31,000 | 50,000 |
| Update GasPriceOracle Proxy | 31,000 | 50,000 |
| Enable Ecotone | 58,000 | 80,000 |
